.github/workflows/test-gcp-dm.yml

```diff
@@ -0,0 +1,120 @@
+name: GCP Deployment Manager Test
+
+on:
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+"
+    types: [opened, synchronize, reopened]
+    paths:
+      - "deploy/deployment-manager/compute_engine.py"
+      - "deploy/deployment-manager/compute_engine.py.schema"
+      - "deploy/deployment-manager/deploy.sh"
+      - "deploy/deployment-manager/set_env.sh"
+
+env:
+  WORKING_DIR: deploy/test-environments
+  GCP_DEPLOYMENT_NAME: ea-cspm-ci-dm-test
+  TF_VAR_ec_api_key: ${{ secrets.EC_API_KEY }}
+  TF_VAR_ess_region: gcp-us-west2 # default region for testing deployments
+  TF_VAR_deployment_name: gcp-dm-ci-test
```

Comment:
Although `ec_project` modifies `deployment_name` by adding an additional suffix to `project_name`, it might be better to generate `TF_VAR_deployment_name` as well. This is because the` test-environments` project is independent, and if someone decides to remove randomization, we may not be able to track the dependency easily. This would only be revealed after running this workflow.